File: annotation/src/main/java/com/github/snowdream/toybricks/annotation/InterfaceLoader.java
package com.github.snowdream.toybricks.annotation;

import proguard.annotation.KeepImplementations;
import proguard.annotation.KeepName;

/**
 * Created by snowdream on 17/2/9.
 */
@KeepName
@KeepImplementations
public interface InterfaceLoader {

    <T> T getImplementation(Class<T> clazz);
}

File: annotation/src/main/kotlin/com/github/snowdream/toybricks/annotation/InterfaceLoader.kt
package com.github.snowdream.toybricks.annotation

/**
 * Created by snowdream on 17/2/9.
 */
interface InterfaceLoader {

    /**
     * get Implementation for the interface.
     */
    fun <T> getImplementation(clazz: Class<T>): T?
}
